src/java/org/apache/cassandra/cql3/constraints/FunctionColumnConstraint.java

```diff
@@ -0,0 +1,164 @@
+
+    private FunctionColumnConstraint(ConstraintFunction function, ColumnIdentifier columnName, Operator relationType, String term)
+    {
+        this.function = function;
+        this.columnName = columnName;
+        this.relationType = relationType;
+        this.term = term;
+    }
+
+    @Override
+    public void appendCqlTo(CqlBuilder builder)
+    {
+        builder.append(toString());
+    }
+
+    @Override
+    public IVersionedSerializer<FunctionColumnConstraint> serializer()
+    {
+        return serializer;
+    }
+
+    @Override
+    public void evaluate(Class<? extends AbstractType> valueType, Object columnValue)
+    {
+        function.evaluate(valueType, relationType, term, columnValue);
+    }
+
+    @Override
+    public void validate(ColumnMetadata columnMetadata)
+    {
+        validateArgs(columnMetadata);
+        function.validate(columnMetadata);
+    }
+
+    @Override
+    public ConstraintType getConstraintType()
+    {
+        return ConstraintType.FUNCTION;
+    }
+
+    void validateArgs(ColumnMetadata columnMetadata)
+    {
+        if (!columnMetadata.name.equals(columnName))
+            throw new InvalidConstraintDefinitionException("Function parameter should be the column name");
+    }
+
+    @Override
+    public String toString()
+    {
+        return function.getName() + "(" + columnName + ") " + relationType + " " + term;
+    }
+
+    public static class Serializer implements IVersionedSerializer<FunctionColumnConstraint>
+    {
+        @Override
+        public void serialize(FunctionColumnConstraint columnConstraint, DataOutputPlus out, int version) throws IOException
+        {
+            out.writeUTF(columnConstraint.function.getName());
+            out.writeUTF(columnConstraint.columnName.toCQLString());
+            out.writeUTF(columnConstraint.relationType.toString());
+            out.writeUTF(columnConstraint.term);
+        }
+
+        @Override
+        public FunctionColumnConstraint deserialize(DataInputPlus in, int version) throws IOException
+        {
+            String functionName = in.readUTF();
+            ConstraintFunction function;
+            String columnNameString = in.readUTF();
+            ColumnIdentifier columnName = new ColumnIdentifier(columnNameString, true);
+            try
+            {
+                function = createConstraintFunction(functionName, columnName);
+            }
+            catch (Exception e)
+            {
+                throw new IOException(e);
+            }
+            String relationTypeString = in.readUTF();
+            Operator relationType = Operator.valueOf(relationTypeString);
+            final String term = in.readUTF();
+            return new FunctionColumnConstraint(function, columnName, relationType, term);
+        }
+
+        @Override
+        public long serializedSize(FunctionColumnConstraint columnConstraint, int version)
+        {
+            FunctionColumnConstraint condition = columnConstraint;
```

Comment:
Can remove redundant aliased variable here and just use `columnConstraint` directly.